extension-manifest-v3/src/background/reporting.js

```diff
@@ -216,26 +216,39 @@ const reporting = new Reporting({
   communication,
 });
 
-observe('terms', (terms) => {
-  if (terms) {
-    reporting.init().catch((e) => {
-      console.warn(
-        'Failed to initialize reporting. Leaving the module disabled and continue.',
-        e,
-      );
-    });
-  } else {
-    reporting.unload();
-  }
+const pendingStartup = new Promise((done) => {
+  let timeout = setTimeout(() => {
+    console.warn('Startup takes too long. Waking up pending listeners...');
+    timeout = null;
+    done();
+  }, 3000);
+
+  observe('terms', async (terms) => {
+    try {
+      if (terms) {
+        try {
+          await reporting.init();
+        } catch (e) {
+          console.warn(
+            'Failed to initialize reporting. Leaving the module disabled and continue.',
+            e,
+          );
+        }
+      } else {
+        reporting.unload();
+      }
+    } finally {
+      clearTimeout(timeout);
```

Comment:
The observer is called multiple times, if the background process starts with `terms: false` and then `terms: true` - for example a usual case when a user installs an extension. I think `done()` nor `clearTimeout` should not be called multiple times.

Reply:
Since the timer is set to "null", calling clearTimeout should be safe. Calling resolve (the "done" function") multiple times is also safe: the first call wins and following calls will be ignored by Promises that are already resolved.

From pure correctness, I should the current version is safe. Unless changing has other benefits that I'm missing?